File: daq/slc/apps/runcontrold/src/RunConfig.cc
#include "daq/slc/apps/runcontrold/RunConfig.h"

#include "daq/slc/apps/runcontrold/RCMaster.h"

using namespace Belle2;

RunConfig::RunConfig(const std::string& data_name, int revision) throw()
  : DataObject(data_name, "RunConfig"), _master(NULL)
{
  setRevision(revision);
  addText("run_type", "TESTT:DEFAULT", 64);
  addText("description", "Defualt settings", 128);
}

void RunConfig::add(std::map<std::string, DataObject*>& data_m,
                    std::vector<NSMNode*>& node_v)
{
  for (std::map<std::string, DataObject*>::iterator it = data_m.begin();
       it != data_m.end(); it++) {
    if (it->first.size() > 0) {
      DataObject* data = it->second;
      std::string label = it->first;;
      addInt(label, data->getConfigNumber());
      _data_m.insert(std::map<std::string, DataObject*>::value_type(label, data));
    }
  }
  for (std::vector<NSMNode*>::iterator it = node_v.begin();
       it != node_v.end(); it++) {
    NSMNode* node = *it;
    addBool(node->getName() + "_used", node->isUsed());
  }
}

void RunConfig::update()
{
  for (std::map<std::string, DataObject*>::iterator it = _data_m.begin();
       it != _data_m.end(); it++) {
    if (it->first.size() > 0) {
      DataObject* data = it->second;
      std::string label = it->first;
      setInt(label, data->getConfigNumber());
    }
  }
  if (_master != NULL) {
    RCMaster::NSMNodeList& node_v(_master->getNSMNodes());
    for (RCMaster::NSMNodeList::iterator it = node_v.begin();
         it != node_v.end(); it++) {
      NSMNode* node = *it;
      setBool(node->getName() + "_used", node->isUsed());
      if (hasValue(node->getName()) && node->getData() != NULL) {
        setInt(node->getName(), node->getData()->getConfigNumber());
      }
    }
  }
}

File: daq/slc/extra/javalib/b2rc/src/b2rc/core/RunConfig.java
package b2rc.core;

import java.util.ArrayList;
import java.util.HashMap;

public class RunConfig extends DataObject {
	
	private HashMap<String, DataObject> _data_m = new HashMap<String, DataObject>();
	private RCMaster _master = null;

	public RunConfig() {
		this("", 0);
	}

	public RunConfig(String data_name, int revision) {
		super(data_name, "RunConfig");
		setRevision(revision);
		addText("run_type", "TESTT:DEFAULT", 64);
		addText("description", "Defualt settings", 128);
	}

	public String getRunType() {
		return getText("run_type");
	}

	public String getDescription() {
		return getText("description");
	}

	public void setRunType(String run_type) {
		setText("run_type", run_type);
	}

	public void setDescription(String description) {
		setText("_description", description);
	}

	public void add(HashMap<String, DataObject> data_m,
					  ArrayList<RCNode> node_v) {
		for (String name : data_m.keySet()) {
			if (name.length() > 0) {
				DataObject data = data_m.get(name);
				String label = name;
				addInt(label, data.getConfigNumber());
				_data_m.put(label, data);
			}
		}
		for ( RCNode node : node_v ) {
			addBool(node.getName() + "_used", node.isUsed());
		}
	}

	public void update() {
		for (String name : _data_m.keySet()) {
			if (name.length() > 0) {
				DataObject data = _data_m.get(name);
				String label = name;
				setInt(label, data.getConfigNumber());
			}
		}
		if (_master != null) {
			for (RCNode node : _master.getNSMNodes()) {
				setBool(node.getName() + "_used", node.isUsed());
				if (hasValue(node.getName()) && node.getData() != null) {
					setInt(node.getName(), node.getData().getConfigNumber());
				}
			}
		}
	}

	public void setMaster(RCMaster master) {
		_master = master;
	}

}
